src/script/interpreter.h

```diff
@@ -166,6 +166,17 @@ class BaseSignatureChecker
     virtual ~BaseSignatureChecker() {}
 };
 
+/** A low level signature checker.  Note that the caller must verify that the hash passed in relates to a known message (unlike for the other signature checkers). */
+class SimpleSignatureChecker : public BaseSignatureChecker
```

Comment:
I've done a proposal to change from a special signature checker to reusing transaction signature checking at https://github.com/bitcoin/bips/pull/947 . @sdaftuar I think this might resolve your concern from https://github.com/bitcoin/bitcoin/pull/16653#issuecomment-523495761